vj4/handler/problem.py

```diff
@@ -76,7 +78,24 @@ class ProblemDetailView(base.Handler):
   @base.sanitize
   async def post(self, *, pid: document.convert_doc_id, lang: str, code: str):
     pdoc = await problem.get(self.domain_id, pid)
-    rid = await record.add(self.domain_id, pdoc['doc_id'], self.user['_id'], lang, code)
+    rid = await record.add(self.domain_id, pdoc['doc_id'], record.TYPE_SUBMISSION, self.user['_id'], lang, code)
+    await asyncio.gather(queue.publish('judge', rid=rid), bus.publish('record_change', rid))
+    self.json_or_redirect(self.reverse_url('record_main'))
+
+
+@app.route(r'/p/{pid}/pretest', 'problem_pretest')
```

Comment:
Either change the view name to `problem_test` or change the class name to `ProblemPretestView`.